### PR TITLE
Added Team Timestamp History + Robot state updating bug fix

### LINKED ISSUE #424: Add a "most_recent_update" Timestamp To Team
### Description of the task
Add a `Timestamp` to the `Team` class that represents the last time that that class was updated.

### Outline the task in more detail
We should add a `Timestamp` class to the `Team` class that represents the last time that the Team was updated. When the `Team` is updated (either the entire state or just some robots), we should update the Team Timestamp to match the new information. 

If the new information has a Timestamp older then the Team timestamp we should throw an exception indicating as such.

### Acceptance criteria

- [ ] Add a "last_time_updated" member variable and getter function to the `Team` class
- [ ] When the team is updated or specific robots are updated, update the "last_time_updated"
- [ ] When we try to update a team with information older then the "last_time_updated", throw an exception
- [ ] Unit tests for all aforementioned functionality!

### PR DESCRIPTION
**Pull request by EvanMorcom**
### Description

This PR includes changes to allow a Timestamp history for the Team object.
Additionally, includes a fix to the Robot class where the state history is transferred in reverse if a Robot is updated using another Robot.

### Testing Done

Unit tests for each new function.

### Resolved Issues

Resolves #424 
- [X] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [X] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [X] **Remove all commented out code**
- [X] **Remove extra print statements**: for example, those just used for testing
- [X] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [X] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by EvanMorcom** [comment]
Addidently added a whole bunch of generated files. Working on deleting them.

File: src/thunderbots/software/ai/world/robot.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ void Robot::updateState(const Robot &new_robot_data)
             "Error: Robot updated using a robot with a mismatched id");
     }
 
-    for (int i = 0; i < new_robot_data.positions_.size(); i++)
+    for (int i = new_robot_data.positions_.size() - 1; i >= 0; i--)
```
Comment (EvanMorcom) [inline src/thunderbots/software/ai/world/robot.cpp:41]: The issue here was that the top item of the queue is the most recent, and this code places the newest item at the back of the new queue.

File: src/thunderbots_msgs/msg/Team.msg [hunk 2]
```diff
@@ -10,3 +10,6 @@ int32 goalie_id
 # The number of milliseconds a robot must not have
 # been updated for before it is removed from the team
 float64 robot_expiry_buffer_milliseconds
+
+# The timestamp of this data in seconds
+float64 timestamp_seconds
```
Comment (MathewMacDougall) [inline src/thunderbots_msgs/msg/Team.msg]: nit: no newline at end of file (GitHub complaining)

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
just one super duper tiny nit

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Comment by garethellis0** [comment]
I fixed the CI issues in another PR (#613). Please merge `master` into this branch (or rebase onto `master`) and CI should pass. Sorry about that!

**Comment by MathewMacDougall** [comment]
@EvanMorcom 	can you update this so we can merge?

File: src/thunderbots_msgs/msg/Team.msg (hunk 2 above)
Comment (EvanMorcom) [reply src/thunderbots_msgs/msg/Team.msg]: Added newline

**Pull request merged**